File: 2398-check-if-matrix-is-x-matrix/2398-check-if-matrix-is-x-matrix.c
bool checkXMatrix(int** grid, int gridSize, int* gridColSize)
{ 
for (int i = 0; i < gridSize; i++) {
        for (int j = 0; j < gridColSize[i]; j++) {
            if (i == j || j == gridSize - 1 - i) {
                if (grid[i][j] == 0) {
                    return false;
                }
            } else {
                if (grid[i][j] != 0) {
                    return false;
                }
            }
        }
    }
    return true;
}

File: 2398-check-if-matrix-is-x-matrix/2398-check-if-matrix-is-x-matrix.py
class Solution(object):
    def checkXMatrix(self, grid):
        flag=0
        n=len(grid)
        for i in range(n):
            for j in range(n):
                if i==j or j==n-i-1:
                    if grid[i][j]==0:
                        flag=1
                        break
                else:
                    if grid[i][j]!=0:
                        flag=1
                        break
        return flag==0